File: LPIS_1/src/main/java/com/example/lab/statement/AssignTarget.java
package com.example.lab.statement;

import com.example.lab.Type;
import com.example.lab.expression.Expression;
import com.example.lab.interpreter.InterpreterContext;

public interface AssignTarget extends Expression {

    boolean init(StatementContext context, Type type);

    void assign(InterpreterContext context, Expression expr);

}

File: LPIS_1/src/main/java/com/example/lab/expression/AssignTarget.kt
package com.example.lab.expression

import com.example.lab.interpreter.InterpreterContext
import com.example.lab.statement.StatementContext

sealed interface AssignTarget : Expression {

	fun init(context: StatementContext, type: Expression.Type): Boolean
}

interface ListAssignTarget : ListExpression, AssignTarget {

	fun assign(context: InterpreterContext, value: List<Int>)
}

interface ElementAssignTarget : ElementExpression, AssignTarget {

	fun assign(context: InterpreterContext, value: Int)
}